Bugsnag/Client/BugsnagClient.m

```diff
@@ -559,7 +563,12 @@ - (void)computeDidCrashLastLaunch {
             unlink(crashSentinelPath);
         }
     }
-    BOOL didCrash = handledCrashLastLaunch || didOOMLastLaunch;
+    
+    BOOL shouldReportOOM = NO;
+    BOOL didCrash = (handledCrashLastLaunch ||
+                     [self lastRunEndedWithAppHang] ||
+                     (shouldReportOOM = [self shouldReportOOM]));
```

Comment:
This changes the behaviour of this section due to the assignment side-effect. If either `handledCrashLastLaunch` or `[self lastRunEndedWithAppHang]` is true, `shouldReportOOM = [self shouldReportOOM]` won't be evaluated.

Reply:
That was intentional - we should not report an OOM if a crash or app hang was detected.

Maybe this needs needs more refactoring... I was trying to minimise the number of changes but the code could be made clearer.

Reply:
If it's just temporary and will go away within this batch of changes, I'm OK with it, but otherwise expressions with side effects are just bugs waiting to pounce once someone unfamiliar with this section (or forgets to look for it) attempts to make a change.

Reply:
This logic has been revised to (hopefully) make it more explicit and obvious.